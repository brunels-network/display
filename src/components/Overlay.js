import React from "react";
import PropTypes from "prop-types";

import styles from "./Overlay.module.css";

// Based on https://stackoverflow.com/a/55003301
function Overlay(props) {
  const { useState } = React;
  const [open, setOpen] = useState(true);
  if (open) {
    return (
      <div
        className={styles.overlay}
        onClick={() => {
          props.toggleOverlay();
          setOpen(false);
        }}
      >
        <div
          onClick={(e) => {
            e.stopPropagation();
          }}
        >
          {props.children}
        </div>
      </div>
    );
  }
  return null;
}

export default Overlay;

Overlay.propTypes = {
  toggleOverlay: PropTypes.func.isRequired,
  children: PropTypes.element.isRequired,
};
