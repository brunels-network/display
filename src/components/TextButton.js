
import PropTypes from "prop-types";
import React from "react";

import styles from "./TextButton.module.css";


class TextButton extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {

    let onClick = null;

    if (this.props.onClick) {
      onClick = this.props.onClick;
    } else {
      onClick = () => null;
    }

    return (
      <button className={styles.button} onClick={onClick}>
        {this.props.children}
      </button>
    );
  }
}

TextButton.propTypes = {
  children: PropTypes.any,
  onClick: PropTypes.func,
};

export default TextButton;
