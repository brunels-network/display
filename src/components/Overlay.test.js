import React from "react";

import { render, cleanup, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Overlay from "./Overlay";
import TextButton from "./TextButton";

afterEach(cleanup);

describe("Overlay Item", () => {
  it("Text on overlay renders", () => {
    const toggleOverlayFn = jest.fn();
    const testString = "This is a test";
    render(
      <Overlay toggleOverlay={toggleOverlayFn} useBackground={true}>
        <TextButton>{testString}</TextButton>
      </Overlay>
    );

    expect(screen.queryAllByText(testString)).toBeTruthy();
  });

  it("With background selects correct background", () => {
    const toggleOverlayFn = jest.fn();
    const testString = "This is a test";
    render(
      <Overlay toggleOverlay={toggleOverlayFn} useBackground={false}>
        <TextButton>{testString}</TextButton>
      </Overlay>
    );

    expect(screen.queryByTestId("overlay")).toHaveStyle(`background: transparent`);
  });
});
