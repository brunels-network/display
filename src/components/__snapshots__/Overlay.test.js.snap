// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Item No background matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="overlayNoBackground"
    data-testid="overlay"
  >
    <div>
      <button
        class="button"
      >
        This is a test
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Overlay Item With background matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="overlay"
    data-testid="overlay"
  >
    <div>
      <button
        class="button"
      >
        This is a test
      </button>
    </div>
  </div>
</DocumentFragment>
`;
